service/history/workflow/update/registry.go

```diff
@@ -178,6 +181,14 @@ func (r *RegistryImpl) HasUndeliveredUpdates() bool {
 	return false
 }
 
+func (r *RegistryImpl) CancelUpdates(ctx context.Context, eventStore EventStore) {
```

Comment:
Needs a function comment. The impl should copy the map before iterating so that it can drop the lock before calling Update.cancel